front-end/src/components/PostListItem.tsx

```diff
@@ -25,14 +28,52 @@ export default function PostListItem(props:Props) {
   const [isModalOpen, setIsModalOpen] = useState(false);
   const [isEditModalOpen, setIsEditModalOpen] = useState(false);
   const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  const [likes, setLikes] = useState<Like[]>([]);
+  const [hasLiked, setHasLiked] = useState<boolean>(false);
 
   const toggle = () => setIsModalOpen(!isModalOpen);
   const toggleEdit = () => setIsEditModalOpen(!isEditModalOpen);
   const toggleDelete = () => setIsDeleteModalOpen(!isDeleteModalOpen);
 
-  const EditCardLink = () => props.loggedInUser !== undefined && isAuthorPost ? <CardLink onClick={()=>{setIsEditModalOpen(true);}}>Edit</CardLink> : null
-  const DeleteCardLink = () => props.loggedInUser !== undefined && isAuthorPost ? <CardLink onClick={()=>{setIsDeleteModalOpen(true);}}>Delete</CardLink> : null
- 
+  useEffect(() => {
+    axios.get(`${process.env.REACT_APP_API_URL}/api/author/${post.author.id}/posts/${post.id}/likes/`).then(res => {
+      const resLikes: Like[] = res.data.items;
+      setLikes(resLikes);
+      setHasLiked(resLikes.filter((l:Like)=>l.author.id===props.loggedInUser?.authorId).length !== 0);
+    })
+  }, [hasLiked]);
+
+  function likePost():void {
+    if(props.loggedInUser){
+      axios.get(`${process.env.REACT_APP_API_URL}/api/auth-user/${props.loggedInUser.username}`)
+            .then( res => {
+                const author: Author = res.data;
+                const like: Like = {
+                  type: 'like',
+                  author: author,
+                  object: `${process.env.REACT_APP_API_URL}/api/author/${props.post.author.id}/posts/${props.post.id}`
+                }
+                axios.post(`${process.env.REACT_APP_API_URL}/api/author/${props.post.author.id}/inbox/`,
+                  like
+                ).then( res => {
+                  alert('You liked the post!');
+                  setHasLiked(true);
+                });
+              }
+            )
+    }else{
```

Comment:
Another nit; but maybe just do the below to avoid unnecessary nesting / indentation?
```javascript
if (!props.loggedInUser) {
  console.error(...)  
  return;
}

// make requests here
```